typing/types.ml

```diff
@@ -418,6 +418,28 @@ let equal_tag t1 t2 =
   | Extension (path1,_), Extension (path2,_) -> Path.same path1 path2
   | (Ordinary _ | Extension _), _ -> false
 
+let equal_variant_representation r1 r2 = r1 == r2 || match r1, r2 with
```

Comment:
Why short-circuit here but not in `equal_record_representation`? I really don't know the value of these short-circuit checks -- but we should be principled about them, either doing them everywhere, nowhere, or informed by perf checks.